scripts/base36-uuid.py

```diff
@@ -0,0 +1,216 @@
+        timestamp = decimicrosecs + int(delta.total_seconds()) * DECIMICRO_RATIO
+        msb = TimeUuid._time_to_msb(timestamp)
+        return TimeUuid(msb, lsb)
+
+    def encode_with_base36(self) -> str:
+        '''encode an integer using base36 representation'''
+        seconds, decimicro = divmod(self.uuid.time, DECIMICRO_RATIO)
+        delta = datetime.timedelta(seconds=seconds)
+        encoded_days = encode(delta.days)
+        encoded_seconds = encode(delta.seconds)
+        encoded_decimicro = encode(decimicro)
+        encoded_lsb = encode(self.lsb)
+        return (f'{encoded_days:0>4}_'
+                f'{encoded_seconds:0>4}_'
+                f'{encoded_decimicro:0>5}'
+                f'{encoded_lsb:0>13}')
+
+    @staticmethod
+    def _time_to_msb(time: int) -> int:
+        # see https://datatracker.ietf.org/doc/html/rfc4122.html
+        time_low = (2 ** 32 - 1) & time
+        time >>= 32
+        time_mid = (2 ** 16 - 1) & time
+        time >>= 16
+        if time >> 12 != 0:
+            raise ValueError(f'time "{time:#016x}" is too large to fit in')
+        time_hi = (2 ** 12 - 1) & time
+        # sets the version to 1
+        time_hi_version = 1 << 12 | time_hi
+        return (time_low << 32 |
+                time_mid << 16 |
+                time_hi_version)
+
+    @property
+    def msb(self) -> int:
+        return (self.uuid.time_low << 32 |
+                self.uuid.time_mid << 16 |
+                self.uuid.time_hi_version)
+
+    @property
+    def lsb(self) -> int:
+        return (self.uuid.clock_seq_hi_variant << 56 |
+                self.uuid.clock_seq_low << 48 |
+                self.uuid.node)
+
+    @property
+    def timestamp(self) -> (datetime.datetime, int):
+        seconds, decimicro_seconds = divmod(self.uuid.time, DECIMICRO_RATIO)
+        return datetime.datetime.fromtimestamp(seconds), decimicro_seconds
+
+    def print_field(self, field: str, print_in_hex: bool) -> None:
+        def print_num(n: int, bits: int) -> str:
+            if print_in_hex:
+                # each hex char represents 4 bits
+                width = bits // 4
+                print(f'{field} = 0x{n:0{width}x}')
+            else:
+                print(f'{field} = {n}')
+
+        if field == 'lsb':
+            print_num(self.lsb, 64)
+        elif field == 'msb':
+            print_num(self.msb, 64)
+        elif field == 'date':
+            datetime, _ = self.timestamp
+            print(f'date = {datetime}')
+        elif field == 'decimicro_seconds':
+            _, decimicro_seconds = self.timestamp
+            print_num(decimicro_seconds, DECIMICRO_RATIO.bit_length())
+        elif field == 'time':
+            print_num(self.uuid.time, 60)
+        elif field == 'node':
+            print_num(self.uuid.node, 48)
+        else:
+            assert False, f'unknown field: {field}'
+
+    def __str__(self) -> str:
+        return self.encode_with_base36()
+
+
+def test_dencode_base36() -> None:
```

Comment:
How is this test run?
You can add a command line option to run it.

Reply:
`pytest base36-uuid.py`

this is not part of a public interface for end user. And IMHO, we should not assume the existence of pytest.

Reply:
I would never guess you run this with pytest in this way although I guess the comment "a minimal pytest" should have hinted that. Maybe be more explicit - "A minimal test, run it with 'pytest base36-uuid.py'"?

Reply:
sure.